src/runtime_src/core/pcie/driver/linux/xocl/subdev/xmc.c

```diff
@@ -3201,7 +3202,7 @@ static void xmc_enable_mailbox(struct xocl_xmc *xmc)
 
 	xmc->mbx_enabled = true;
 	safe_read32(xmc, XMC_HOST_MSG_OFFSET_REG, &val);
-	xmc->mbx_offset = val;
+	xmc->mbx_offset = val ? val : XMC_MBX_DEF_OFFSET;
```

Comment:
Is this temporary workaround or permanent change? I would think for some reason the data in the mailbox is corrupted or temporary unavailable.

Reply:
> 
> 
> Is this temporary workaround or permanent change? I would think for some reason the data in the mailbox is corrupted or temporary unavailable.

had a offline chat with @xuhz. he will remove this hard-code offset and do it differently.

Reply:
it is a common issue for all cases where cmc is running on ps. so far there is no dependency for xmc driver loading on host and cmc running on ps. So when xmc is probed, the cmc may not be ready yet.
old u30 doesn't have POR, so after reset, the cmc is up and running pretty fast. On new u30, it takes ~1min for the ps up, before that xmc on host may have been probed and a 0 is saved as mailbox offset.
Don't know versal has same case or not

Reply:
Versal boot time is around 10s as of today. I think return EIO and user retry the flash is safer.